core/src/main/scala/chisel3/Data.scala

```diff
@@ -663,7 +580,8 @@ abstract class Data extends HasId with NamedComponent with SourceInfoDoc {
           )
       }
     } else {
-      this.legacyConnect(that)
+      if(connectCompileOptions.migrateConnections) Builder.error(s"Cannot use <> to connect Chisel bundles; use :<>= instead")
```

Comment:
I don't think this is the right error  message. This is not about connecting Chisel bundles but about the Chisel file where the <> is called. Can we change this message to reflect that?